src/EventListener/ExceptionConverterListener.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\InternalServerErrorHttpException;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * Converts exceptions to HTTP exceptions.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+class ExceptionConverterListener
+{
+    /**
+     * @var array
+     */
+    private $mapper = [
+        'Contao\\CoreBundle\\Exception\\AccessDeniedException'           => 'AccessDeniedHttpException',
+        'Contao\\CoreBundle\\Exception\\ForwardPageNotFoundException'    => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\IncompleteInstallationException' => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\InsecureInstallationException'   => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\InvalidRequestTokenException'    => 'BadRequestHttpException',
+        'Contao\\CoreBundle\\Exception\\NoActivePageFoundException'      => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\NoLayoutSpecifiedException'      => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\NoRootPageFoundException'        => 'InternalServerErrorHttpException',
+        'Contao\\CoreBundle\\Exception\\PageNotFoundException'           => 'NotFoundHttpException',
+        'Contao\\CoreBundle\\Exception\\ServiceUnavailableException'     => 'ServiceUnavailableHttpException',
+    ];
+
+    /**
+     * Adds the referer ID to the request.
```

Comment:
I know and I suggested to use "Maps known exceptions to HTTP exceptions" instead. You like?